src/main.ts

```diff
@@ -10,16 +10,244 @@
+        .option('out', {
+          describe: 'Manifest file name',
+          requiresArg: true,
+          string: true
+        })
+        .option('name', {
+          describe: 'Contract name',
+          requiresArg: true,
+          string: true
+        })
+        .alias('n', 'name')
+        .option('slim', {
+          describe: 'Slim contract bundle',
+          requiresArg: true,
+          string: true
+        })
+        .alias('s', 'slim')
+        .option('version', {
+          describe: 'Contract version',
+          requiresArg: true,
+          string: true
+        })
+        .alias('v', 'version')
+        .positional('signingKey', {
+          describe: 'Signing key file',
+          demandOption: true,
+          type: 'string'
+        })
+        .positional('contractBundle', {
+          describe: 'Contract bundle',
+          demandOption: true,
+          type: 'string'
+        })
+    }, (argv) => {
+      return commands.manifest(argv)
+    })
+    .command('migrate [--to backend]', '', (yargs) => {
+      return yargs
+        .option('to', {
+          describe: 'Destination backend',
+          demandOption: true,
+          requiresArg: true,
+          string: true
+        })
+    }, (argv) => {
+      return commands.migrate(argv)
+    })
+    .command('serve [--port PORT] [--dashboard-port PORT] [directory]', 'start the server', (yargs) => {
+      return yargs
+        .option('port', {
+          default: 8000,
+          describe: 'Port to listen on (app)',
+          requiresArg: true,
+          number: true
+        })
+        .alias('p', 'port')
+        .option('dashboard-port', {
+          default: 8888,
+          describe: 'Port to listen on (dashboard)',
+          requiresArg: true,
+          number: true
+        })
+        .alias('d', 'dashboard-port')
+        .positional('directory', {
+          default: '.',
+          describe: 'Directory',
+          type: 'string'
+        })
+    }, (argv) => {
+      return commands.serve(argv)
+    })
+    .command('upload [--url REMOTE_URL] <files..>', '', (yargs) => {
+      return yargs
+        .option('url', {
+          describe: 'URL of a remote server',
+          requiresArg: true,
+          string: true
+        })
+        .positional('files', {
+          describe: 'Files to upload',
+          demandOption: true,
+          array: true,
+          type: 'string'
+        })
+    }, (argv) => {
+      return commands.upload(argv)
+    })
+    .command('verifySignature <manifestFile>', '', (yargs) => {
+      return yargs
+        .option('key', {
+          describe: 'Public key',
+          requiresArg: true,
+          string: true,
+        })
+        .alias('k', 'key')
+        .positional('manifestFile', {
+          describe: 'Manifest file',
+          demandOption: true,
+          type: 'string'
+        })
+    }, (argv) => {
+      return commands.verifySignature(argv)
+    })
+    .help()
+  )
```

Comment:
It would be greatly preferable for each command to setup its own arguments in its own file. `main.ts` is intended to be a very small compact delegation mechanism. Can you see if there's a way to perhaps pass `nconf` to the corresponding commands to configuration?